### PR TITLE
document the singleProcessOOMKill kubelet flag

### PR DESCRIPTION
**Pull request by tzneal**
Documents the new kubelet flag being added by https://github.com/kubernetes/kubernetes/pull/122813

### CODE REVIEW

File: content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 1]
```diff
@@ -363,6 +363,13 @@ relative to their scheduling requests are killed first.
 Unlike pod eviction, if a container is OOM killed, the kubelet can restart it
 based on its `restartPolicy`.
 
+{{<note>}}
+If using [cgroup v2](/docs/concepts/architecture/cgroups/), the kubelet sets the `memory.oom.group` flag so that if a single process in
+a container is OOM killed, all processes in the container will be killed as a group. You can set
+the `singleProcessOOMKill` flag in the [kubelet configuration file](/docs/tasks/administer-cluster/kubelet-config-file/) to
+cause kubelet to retain the previous behavior when using cgroup v2.
```
Comment (sftim) [inline content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md:370]:
```suggestion
cause kubelet to select the legacy kubelet behavior when using cgroup v2.
```
“retain” implies that the cluster was deployed with an older version and then upgraded, which we can't assume.

**Review by sftim — COMMENTED** [review]
Thanks for the PR.

Will this change be released as part of v1.29 or for v1.30?

If it's a v1.30 change, the PR needs to be against _dev-1.30_ instead. We continuously deploy the docs but only merge in docs for a release [moments] after the binaries are shipped.

/sig node

File: content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md (hunk 1 above)
Comment (tzneal) [reply content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md]: SGTM

**Comment by tzneal** [comment]
> Thanks for the PR.
> 
> Will this change be released as part of v1.29 or for v1.30?
> 
> If it's a v1.30 change, the PR needs to be against _dev-1.30_ instead. We continuously deploy the docs but only merge in docs for a release [moments] after the binaries are shipped.
> 
> /sig node

It will be in 1.30 first, it may be backported to other releases.  I re-targeted the dev-1.30 branch for now.

**Comment by sftim** [comment]
LGTM for docs

**Comment by divya-mohan0209** [comment]
FYA @drewhagen 

@kubernetes/sig-node-pr-reviews : PTAL

**Comment by Vyom-Yadav** [comment]
Hey @tzneal 👋 please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday March 12th 2024 18:00 PST. Thank you!

---

This has lgtm label and looks ready, so just a friendly reminder :)

**Comment by Vyom-Yadav** [comment]
Hey @tzneal 👋🏼, I see no recent activity in the [linked PR](https://github.com/kubernetes/kubernetes/pull/122813). We are way past the code freeze, and I don't think that PR will get merged now.

Removing from the current milestone, you can get this merged in the next cycle if the changes are good in this PR.

/milestone clear

If you think that PR will still get merged, please get it technically reviewed by March 26th 2024 18:00 PT to get this into the release. Thank you!

**Comment by drewhagen** [comment]
/hold
@tzneal Kubernetes 1.30 is releasing today.
This PR will need to be targeting the `dev-1.31` branch once it is available. Our SIG Release docs team will keep you posted.

**Comment by drewhagen** [comment]
You can reopen this in the next release cycle and target `dev-1.31`. Thanks! :)

**Pull request closed without merging**